cloudify_types/cloudify_types/component/component.py

```diff
@@ -230,6 +231,20 @@ def _set_secrets(self):
             return
 
         for secret_name in self.secrets:
+            existing_secret_value = get_secret_by_name(self.client,
+                                                       secret_name)
+            if existing_secret_value:
+                if self.secrets[secret_name] == existing_secret_value:
+                    ctx.logger.info('Not creating secret "{0}" because it is '
+                                    'already exists with same value...'.format(
+                                     secret_name))
+                    continue
+                else:
+                    raise NonRecoverableError('Tried to upload a secret "{0}"'
```

Comment:
Wouldn't it be better to not check the value at all, but instead just check to see if a secret by the same name exists and say "Not modifying pre-existing secret <name>..."?

Reply:
Or even to have a flag that could be provided (e.g. --force to override them or another flag to fail if any already exist)